### PR TITLE
KAFKA-18062: use feature version to enable ELR

### PR DESCRIPTION
**Pull request by CalvinLiu7947**
Replace the ELR static config with feature version.
https://issues.apache.org/jira/browse/KAFKA-18062

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java [hunk 1]
```diff
@@ -1029,7 +1030,8 @@ TimelineHashSet<TopicIdPartition> imbalancedPartitions() {
     }
 
     boolean isElrEnabled() {
-        return eligibleLeaderReplicasEnabled && featureControl.metadataVersion().isElrSupported();
+        return featureControl.metadataVersion().isElrSupported() && featureControl.finalizedFeatures(offsetControl.lastStableOffset()).
```
Comment (cmccabe) [inline metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java]: I don't think this is correct. Why would we want isElrEnabled to be reading from a point in the past? Surely we want to know whether elr is CURRENTLY enabled, right?

File: metadata/src/test/java/org/apache/kafka/controller/QuorumControllerIntegrationTestUtils.java [hunk 2]
```diff
@@ -75,6 +77,27 @@ static BrokerRegistrationRequestData.FeatureCollection brokerFeatures(
...
     }
 
+    /**
+     * Create a broker features collection for use in a registration request. MV and given features are included.
+     *
+     * @param minVersion    The minimum supported MV.
+     * @param maxVersion    The maximum supported MV.
+     * @param featureList   Other features to add.
+     */
+    static BrokerRegistrationRequestData.FeatureCollection brokerFeaturesPlusFeatureVersions(
```
Comment (cmccabe) [inline metadata/src/test/java/org/apache/kafka/controller/QuorumControllerIntegrationTestUtils.java:86]: seems a bit awkward to use. can we just have a function parameter like a map from feature name to max version (and then assume that min versions are all 0?)

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java (hunk 1 above)
Comment (CalvinLiu7947) [reply metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java]: Updated to use the latest finalized version.

File: metadata/src/test/java/org/apache/kafka/controller/QuorumControllerIntegrationTestUtils.java (hunk 2 above)
Comment (CalvinLiu7947) [reply metadata/src/test/java/org/apache/kafka/controller/QuorumControllerIntegrationTestUtils.java:86]: Updated.

**Review by cmccabe — APPROVED** [review]
LGTM

**Comment by CalvinLiu7947** [comment]
@cmccabe We have a green build. Good to merge?

**Pull request merged**